File: scrabble-board-manager/src/main/java/com/aryzko/scrabble/scrabbleboardmanager/interfaces/web/request/BoardRequest.java
package com.aryzko.scrabble.scrabbleboardmanager.interfaces.web.request;

import com.aryzko.scrabble.scrabbleboardmanager.interfaces.web.common.Field;
import com.aryzko.scrabble.scrabbleboardmanager.interfaces.web.common.Rack;
import com.aryzko.scrabble.scrabbleboardmanager.interfaces.web.validator.UUID;
import jakarta.validation.Valid;
import lombok.Data;

import java.util.List;

@Data
public class BoardRequest {
    @UUID
    private String id;
    @Valid
    private List<Field> fields;
    @Valid
    private List<Rack> racks;
}

File: scrabble-ui/src/app/clients/board-manager/model/board-request.ts
import {Field} from "./field";
import {Rack} from "./rack";
import {BoardParameters} from "./board-parameters";

export class BoardReqest {
  public id: string;
  public fields: Field[];
  public racks: Rack[];

  constructor(id: string, fields: Field[], racks: Rack[], ) {
    this.id = id;
    this.fields = fields;
    this.racks = racks;
  }
}
